### PR TITLE
Implemented ExecuteJavascript function

### PR DESCRIPTION
**Pull request by true-zero**

### CODE REVIEW

**Comment by true-zero** [comment]
Not sure what the issue is since I've never used coveralls before.
From googling, I think it is because there is no test "coverage" for the `ExecuteJavascript` function.

File: event.go [hunk 1]
```diff
@@ -49,6 +49,7 @@ type Event struct {
 	Username            string               `json:"username,omitempty"`
 	WindowID            string               `json:"windowId,omitempty"`
 	WindowOptions       *WindowOptions       `json:"windowOptions,omitempty"`
+	Code                string               `json:"code,omitempty"`
```
Comment (asticode) [inline event.go:52]: Could you make sure your attribute doesn't break alphabetical order ?

File: window.go [hunk 2]
```diff
@@ -14,6 +14,7 @@ import (
 const (
 	EventNameWebContentsEventLogin             = "web.contents.event.login"
 	EventNameWebContentsEventLoginCallback     = "web.contents.event.login.callback"
+	EventNameWebContentsExecuteJavaScript      = "web.contents.execute.javascript"
```
Comment (asticode) [inline window.go:17]: I'd rather rename this event and its constant `window.event.web.contents.executed.javascript`

File: window.go [hunk 3]
```diff
@@ -519,3 +521,11 @@ func (w *Window) Unmaximize() (err error) {
 	_, err = synchronousEvent(w.ctx, w, w.w, Event{Name: EventNameWindowCmdUnmaximize, TargetID: w.id}, EventNameWindowEventUnmaximize)
 	return
 }
+
+func (w *Window) ExecuteJavaScript(code string) (err error) {
```
Comment (asticode) [inline window.go:525]: Could you add a test as well? (you can copy the one for unmaximize for instance)

**Comment by true-zero** [comment]
@asticode Could you pull this?

File: window.go (hunk 2 above)
Comment (asticode) [reply window.go:17]: This const should be:

`EventNameWindowEventWebContentsExecutedJavaScript = "window.event.web.contents.executed.javascript"`

File: window.go [hunk 4]
```diff
@@ -34,6 +35,7 @@ const (
 	EventNameWindowCmdUnmaximize               = "window.cmd.unmaximize"
 	EventNameWindowCmdWebContentsCloseDevTools = "window.cmd.web.contents.close.dev.tools"
 	EventNameWindowCmdWebContentsOpenDevTools  = "window.cmd.web.contents.open.dev.tools"
+	EventNameWindowCmdWebContentsExecutedJavaScript = "window.event.web.contents.executed.javascript"
```
Comment (asticode) [inline window.go:38]: This constant should be `EventNameWindowCmdWebContentsExecuteJavaScript = "window.cmd.web.contents.execute.javascript"`

File: window.go [hunk 5]
```diff
@@ -519,3 +521,11 @@ func (w *Window) Unmaximize() (err error) {
 	_, err = synchronousEvent(w.ctx, w, w.w, Event{Name: EventNameWindowCmdUnmaximize, TargetID: w.id}, EventNameWindowEventUnmaximize)
 	return
 }
+
+func (w *Window) ExecuteJavaScript(code string) (err error) {
+	if err = w.ctx.Err(); err != nil {
+		return
+	}
+	_, err = synchronousEvent(w.ctx, w, w.w, Event{Name: EventNameWindowCmdWebContentsExecutedJavaScript, TargetID: w.id, Code: code}, EventNameWebContentsExecuteJavaScript)
```
Comment (asticode) [inline window.go:529]: This should be `_, err = synchronousEvent(w.ctx, w, w.w, Event{Name: EventNameWindowCmdWebContentsExecuteJavaScript, TargetID: w.id, Code: code}, EventNameWindowEventWebContentsExecutedJavaScript)`

File: window_test.go [hunk 6]
```diff
@@ -69,6 +69,7 @@ func TestWindow_Actions(t *testing.T) {
 	testObjectAction(t, func() error { return w.Show() }, w.object, wrt, "{\"name\":\""+EventNameWindowCmdShow+"\",\"targetID\":\""+w.id+"\"}\n", EventNameWindowEventShow)
 	assert.Equal(t, true, w.IsShown())
 	testObjectAction(t, func() error { return w.Unmaximize() }, w.object, wrt, "{\"name\":\""+EventNameWindowCmdUnmaximize+"\",\"targetID\":\""+w.id+"\"}\n", EventNameWindowEventUnmaximize)
+	testObjectAction(t, func() error { return w.ExecuteJavaScript("console.log('test');") }, w.object, wrt, "{\"name\":\""+EventNameWindowCmdWebContentsExecutedJavaScript+"\",\"targetID\":\""+w.id+"\",\"code\":\"console.log('test');\"}\n", EventNameWebContentsExecuteJavaScript)
```
Comment (asticode) [inline window_test.go:72]: This should be `testObjectAction(t, func() error { return w.ExecuteJavaScript("console.log('test');") }, w.object, wrt, "{\"name\":\""+EventNameWindowCmdWebContentsExecuteJavaScript+"\",\"targetID\":\""+w.id+"\",\"code\":\"console.log('test');\"}\n", EventNameWindowEventWebContentsExecutedJavaScript)`

**Comment by asticode** [comment]
FYI I've created the following tags:

- `v0.16.0` for go-astilectron
- `v0.7.0` for go-astilectron-bundler
- `v0.4.1` for go-astilectron-bootstrap
- `v0.6.5` for go-astilectron-demo

**Pull request merged**